socket/src/commonTest/kotlin/mqtt/socket/ServerProcessTest.kt

```
@@ -0,0 +1,104 @@
+import kotlin.time.ExperimentalTime
+import kotlin.time.milliseconds
+
+enum class ServerAction {USHORT, CONNECT_DISCONNECT, MQTTSTRING}
+
+class ServerProcessTest (val action: ServerAction) : TCPServerProcess() {
+    @ExperimentalTime
+    private val timeout = 10000.milliseconds
+    public var name: String = ""
+    public var clientResponse: String = ""
+
+    @ExperimentalUnsignedTypes
+    val limits = object : BufferMemoryLimit {
+        override fun isTooLargeForMemory(size: UInt) = size > 1_000u
+    }
+
+    private suspend fun ConnectProcess () {
+
+    }
+
+    @ExperimentalTime
+    private suspend fun ConnectDisProcess () {
+        val serverReadBuffer = allocateNewBuffer(10.toUInt(), limits)
+
+        try {
+            assertTrue(socket.isOpen(), "socket to client is not open")
+            socket.read(serverReadBuffer, timeout)
+        } catch (e: Exception) {
+            assertEquals("remote returned -1 bytes", e.message, "Unknown exception recevied")
+            socket.close()
+        }
+    }
+
+    @ExperimentalTime
+    private suspend fun UShortProcess() {
+        val serverReadBuffer = allocateNewBuffer(10.toUInt(), limits)
+        val serverWriteBuffer = allocateNewBuffer(10.toUInt(), limits)
+        val recvData: UShort = 4.toUShort()
+        val sendData: UInt = UInt.MAX_VALUE
+
+        try {
+            assertTrue(socket.isOpen(), "socket to client is not open")
+            assertEquals(2, socket.read(serverReadBuffer, timeout), "server received invalid data")
+            assertEquals(recvData, serverReadBuffer.readUnsignedShort(), "server received invalid data")
+            serverWriteBuffer.write(sendData)
+            assertEquals(4, socket.write(serverWriteBuffer, timeout), "server send data size not correct")
+            socket.close()
+        } catch (e: Exception) {
+            println("ServerProcessUshort.exception: ${e.message}, $e")
+            assertEquals("Failure...", e.message)
```

why are we swallowing the exceptions here?